src/foremast/awslambda/awslambda.py

```diff
@@ -54,10 +54,12 @@ def __init__(self, app, env, region, prop_path):
         self.handler = self.pipeline['handler']
         self.vpc_enabled = self.pipeline['vpc_enabled']
 
-        self.memory = self.properties[env]['app']['lambda_memory']
-        self.timeout = self.properties[env]['app']['lambda_timeout']
+        app = self.properties[self.env]['app']
+        self.memory = app['lambda_memory']
+        self.role = app.get('lambda_role') or generated.iam()['lambda_role']
```

Comment:
@e4r7hbug This is causing a bug. We never create `generated.iam()['lambda_role']` and so if no role is specified, it errors out because this lambda specific role isnt created.